daemon/containerd/image_pull.go

```diff
@@ -120,6 +123,9 @@ func (i *ImageService) PullImage(ctx context.Context, ref reference.Named, platf
 
 	img, err := i.client.Pull(ctx, ref.String(), opts...)
 	if err != nil {
+		if errors.Is(err, docker.ErrInvalidAuthorization) {
```

Comment:
This will clash with https://github.com/containerd/containerd/pull/9234

maybe we should wrap the returned error?

Reply:
doh! forgot about that one; I guess we need to revert this one when we bump to a containerd vendor with that patch

Reply:
I guess the check will be even uglier, we'll have to check the error message itself and return the right error

Reply:
Opened: https://github.com/moby/moby/pull/46662